app/models/concerns/orchestration/dhcp.rb

```diff
@@ -123,7 +123,7 @@ def dhcp_attrs(record_mac)
     }
 
     if provision?
-      dhcp_attr[:nextServer] = boot_server
+      dhcp_attr[:nextServer] = boot_server unless self.host.pxe_loader == 'None'
```

Comment:
Shouldn't this just return inside the `boot_server` function?

Reply:
I don't want to set anything, if I do it in `boot_server` the resulting JSON will contain `"nextServer": null` which I don't want to send.

Reply:
But I think the proxy is able to handle that. My concern is that we're missing a good clean way to determine exactly what the TFTP server address will be. On the other hand, the boot_server method isn't used anywhere else either so I guess it's fine.

Reply:
I agree we need to improve TFTP server address detection, that's what I describe in https://community.theforeman.org/t/foreman-and-dhcp-options/15127 but this PR is solely to fix one issue. When loader is set to "None", do not override both nextserver and filename options (we already do the latter).

Tell me what you want me to change and I will do it. Otherwise let's push this there are like 4-5 other PRs in order to improve all of this :-)